tests/MongoDB.Driver.Tests/Specifications/retryable-writes/prose-tests/CommandConstructionTests.cs

```diff
@@ -0,0 +1,332 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers.XunitExtensions;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.Core.Clusters;
+using MongoDB.Driver.Core.Events;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using MongoDB.Driver.TestHelpers;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Specifications.retryable_writes.prose_tests
+{
+    public class CommandConstructionTests
+    {
+        private readonly string _collectionName = CoreTestConfiguration.GetCollectionNamespaceForTestClass(typeof(CommandConstructionTests)).CollectionName;
+        private readonly string _databaseName = CoreTestConfiguration.DatabaseNamespace.DatabaseName;
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Unacknowledged_writes_should_not_have_transaction_id(
+            [Values("delete", "insert", "update")] string operation)
+        {
+            RequireServer.Check().ClusterTypes(ClusterType.ReplicaSet, ClusterType.Sharded);
+
+            DropCollection();
+            var eventCapturer = CreateEventCapturer();
+            using (var client = CreateDisposableClient(eventCapturer))
+            {
+                var database = client.GetDatabase(_databaseName);
+                var collection = database.GetCollection<BsonDocument>(_collectionName);
+
+                switch (operation)
+                {
+                    case "delete":
+                        var deleteFilter = Builders<BsonDocument>.Filter.Eq("_id", 1);
+                        collection.WithWriteConcern(WriteConcern.Unacknowledged).DeleteOne(deleteFilter);
```

Comment:
Nit: Move the repeated calls to WithWriteConcern to line 48 to DRY this up, unless you have a rationale for the repetition.

Reply:
I had that way originally and moved it here to be super explicit that the write operations are unacknowledged.

When it was hidden at the end of line 48 it was too easy to miss.

I can move it if you still want.

As an aside, I think it makes no sense that WriteConcern should be a property to collection. But that's history...

Reply:
Works for me.